### PR TITLE
Adds gradle snapshot deployment

### PR DESCRIPTION
**Pull request by ecgreb**
Deploys a snapshot AAR to Sonatype OSS snapshot repository on each successful build of master.

### CODE REVIEW

**Comment by blair1618** [comment]
Is there a convention of indenting with 2 spaces in gradle files? We use 4 spaces everywhere but it's not a huge deal

**Comment by ecgreb** [comment]
Yeah 2 spaces is the standard indent for gradle/groovy.

File: android/tangram/build.gradle [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
+  }
+}
 
-    compileSdkVersion 15
-    buildToolsVersion "21.1.2"
+apply plugin: 'com.android.library'
+apply plugin: 'com.github.dcendents.android-maven'
 
-    defaultConfig {
-        minSdkVersion 15
```
Comment (tallytalwar) [inline android/tangram/build.gradle]: @ecgreb Don't we have to mention the minSdkVersion and targetVersion?

**Comment by blair1618** [comment]
@ecgreb you may need to update the android platform version in the travis setup script to "22": https://github.com/tangrams/tangram-es/blob/6c17f78970e097ef4fd5511fa58489760d22d6b1/travis/before_install.sh#L48

Travis only installs the parts of the sdk that we need for building, to save time. It's a bit of a pain to configure this in two places, but we don't change it very often.

File: android/tangram/build.gradle (hunk 1 above)
Comment (ecgreb) [inline android/tangram/build.gradle]: Nice catch. Re-added.

**Pull request merged**